core/src/test/scala/pickling/run/non-public-separate.scala

```diff
@@ -7,7 +7,7 @@ class Person(private val name: String, age: Int, val hobby: Hobby) {
   // NOTE: be careful not to reference age anywhere, so that it's elided by the "constructors" phase
   override def toString = s"Person(name = $name, hobby = $hobby)"
 }
-class Hobby(var name: String, private var notes: String, private val attitude: String) {
+final class Hobby(var name: String, private var notes: String, private val attitude: String) {
```

Comment:
OOOH, so this is actually an `isEffectivelyFinal` bug.  It appears to return true for the pickler and not true for the unpickler....